tests/tests.rs

```diff
@@ -1619,3 +1659,30 @@ fn should_restrict_rpc_access() {
         }
     );
 }
+
+#[test]
+fn should_use_custom_response_size_estimate() {
```

Comment:
What about the not-so-happy path?

Reply:
Right, so there's a limitation with `ic-state-machine-tests` where the max HTTP request size unfortunately isn't taken into account. There doesn't seem to be any way to simulate the expected IC error message (which really feels like it should be built into the testing library). For example, if I added a test case which limits the response to zero bytes, the request would succeed the first time. All we can really do is ensure that the custom size estimate is respected when sending the first HTTP outcall.